services/thermal_mgr/thermal_mgr.c

```diff
@@ -1,71 +1,102 @@
-  /* Send an event to the thermal manager queue */
-
-  return ERR_CODE_SUCCESS;
-}
-
-void osHandlerLM75BD(void) {
-  /* Implement this function */
-}
-
-static void thermalMgr(void *pvParameters) {
-  /* Implement this task */
-  while (1) {
-    
-  }
-}
-
-void addTemperatureTelemetry(float tempC) {
-  printConsole("Temperature telemetry: %f deg C\n", tempC);
-}
-
-void overTemperatureDetected(void) {
-  printConsole("Over temperature detected!\n");
-}
-
-void safeOperatingConditions(void) { 
-  printConsole("Returned to safe operating conditions!\n");
-}
+#include "thermal_mgr.h"
+#include "errors.h"
+#include "lm75bd.h"
+#include "console.h"
+#include "../sys/logging/logging.h"
+#include <FreeRTOS.h>
+#include <os_task.h>
+#include <os_queue.h>
+
+#include <string.h>
+//test
+#define THERMAL_MGR_STACK_SIZE 256U
+
+static TaskHandle_t thermalMgrTaskHandle;
+static StaticTask_t thermalMgrTaskBuffer;
+static StackType_t thermalMgrTaskStack[THERMAL_MGR_STACK_SIZE];
+
+#define THERMAL_MGR_QUEUE_LENGTH 10U
+#define THERMAL_MGR_QUEUE_ITEM_SIZE sizeof(thermal_mgr_event_t)
+
+static QueueHandle_t thermalMgrQueueHandle;
+static StaticQueue_t thermalMgrQueueBuffer;
+static uint8_t thermalMgrQueueStorageArea[THERMAL_MGR_QUEUE_LENGTH * THERMAL_MGR_QUEUE_ITEM_SIZE];
+
+static void thermalMgr(void *pvParameters);
+
+void initThermalSystemManager(lm75bd_config_t *config) {
+  memset(&thermalMgrTaskBuffer, 0, sizeof(thermalMgrTaskBuffer));
+  memset(thermalMgrTaskStack, 0, sizeof(thermalMgrTaskStack));
+  
+  thermalMgrTaskHandle = xTaskCreateStatic(
+    thermalMgr, "thermalMgr", THERMAL_MGR_STACK_SIZE,
+    config, 1, thermalMgrTaskStack, &thermalMgrTaskBuffer);
+
+  memset(&thermalMgrQueueBuffer, 0, sizeof(thermalMgrQueueBuffer));
+  memset(thermalMgrQueueStorageArea, 0, sizeof(thermalMgrQueueStorageArea));
+
+  thermalMgrQueueHandle = xQueueCreateStatic(
+    THERMAL_MGR_QUEUE_LENGTH, THERMAL_MGR_QUEUE_ITEM_SIZE,
+    thermalMgrQueueStorageArea, &thermalMgrQueueBuffer);
+
+}
+
+error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
+  /* Send an event to the thermal manager queue */
+  if(thermalMgrQueueHandle == NULL)
+    return ERR_CODE_INVALID_STATE;
+  else if(event == NULL)
+    return ERR_CODE_INVALID_ARG;
+  
+  if(xQueueSend(thermalMgrQueueHandle, (void*) event, 0) == pdPASS)
+    return ERR_CODE_SUCCESS;  
+
+  return ERR_CODE_QUEUE_FULL;
+}
+
+void osHandlerLM75BD(void) {
+  /* Implement this function */
+  thermal_mgr_event_t event;
+  event.type = THERMAL_MGR_EVENT_INTERRUPT;
+  thermalMgrSendEvent(&event);
+}
+
+static void thermalMgr(void *pvParameters) {
+  /* Implement this task */
+  thermal_mgr_event_t eventReceived;
+  error_code_t errCode;
+  while (1) {
+    if(xQueueReceive(thermalMgrQueueHandle, &eventReceived, 10) == pdPASS) {
+      switch (eventReceived.type) {
+        case THERMAL_MGR_EVENT_MEASURE_TEMP_CMD: {
+          float temp;
+          RETURN_IF_ERROR_CODE(readTempLM75BD(LM75BD_OBC_I2C_ADDR, &temp));

```

Comment:
we generally should not return from a FreeRTOS task function so this should be LOG_IF_ERROR_CODE instead. Also RETURN_IF_ERROR_CODE would return the error code, but this is a void function